cmd/aws-k8s-agent/main.go

```diff
@@ -36,44 +38,108 @@ const (
 
 	// Environment variable to disable the IPAMD introspection endpoint on 61679
 	envDisableIntrospection = "DISABLE_INTROSPECTION"
+
+	restCfgTimeout = 5 * time.Second
+	pollInterval   = 5 * time.Second
+	pollTimeout    = 30 * time.Second
 )
 
 func main() {
 	os.Exit(_main())
 }
 
+// startBackgroundAPIServerCheck checks API connectivity in the background
+func startBackgroundAPIServerCheck(ipamContext *ipamd.IPAMContext) {
```

Comment:
For super large clusters, this could lead to hundreds of thousands API server calls. I am not sure if `for` loop calls without timeout is a safe idea.